implementations/rust/ockam/ockam_command/src/enroll/command.rs

```diff
@@ -189,97 +188,89 @@ pub async fn enroll_with_node(
-            *is_finished.lock().await = true;
-            Ok(space)
-        };
-
-        let message = vec![format!(
-            "Creating space {}...",
-            name.color(OckamColor::PrimaryResource.color())
-        )];
-        let progress_output = opts.terminal.progress_output(&message, &is_finished);
-        let (space, _) = try_join!(create_space, progress_output)?;
-        space
-    }
-    // If it has, return the first one on the list
-    else {
-        for space in &available_spaces {
-            opts.state
-                .spaces
-                .overwrite(&space.name, SpaceConfig::from(space))?;
-        }
-
-        let space = available_spaces
-            .drain(..1)
-            .next()
-            .expect("already checked that is not empty");
-
-        opts.terminal.write_line(&fmt_log!(
-            "Found space {}.",
+    let space = match spaces.first() {
+        None => {
+            opts.terminal
+                .write_line(&fmt_para!("No spaces are defined in your account."))?
+                .write_line(&fmt_para!(
+                    "Creating a trial space for you ({}) ...",
+                    "everything in it will be deleted in 15 days"
+                        .to_string()
+                        .color(OckamColor::FmtWARNBackground.color())
+                ))?
+                .write_line(&fmt_para!(
+                    "To learn more about production ready spaces in Ockam Orchestrator, contact us at: {}",
+                    "hello@ockam.io".to_string().color(OckamColor::PrimaryResource.color())))?;
+
+            let is_finished = Mutex::new(false);
+            let space_name = random_name();
+            let create_space = async {
+                let space = node.create_space(ctx, &space_name, vec![]).await?;
+                *is_finished.lock().await = true;
+                Ok(space)
+            };
+
+            let message = vec![format!(
+                "Creating space {}...",
+                space_name
+                    .clone()
+                    .color(OckamColor::PrimaryResource.color())
+            )];
+            let progress_output = opts.terminal.progress_output(&message, &is_finished);
+            let (space, _) = try_join!(create_space, progress_output)?;
             space
-                .name
-                .to_string()
-                .color(OckamColor::PrimaryResource.color())
-        ))?;
-        space
+        }
+        Some(space) => {
+            opts.terminal.write_line(&fmt_log!(
+                "Found space {}.",
+                space
+                    .name
+                    .clone()
+                    .color(OckamColor::PrimaryResource.color())
+            ))?;
+            space.clone()
+        }
     };
-    opts.state
-        .spaces
-        .overwrite(&default_space.name, SpaceConfig::from(&default_space))?;
+    opts.state.set_space_as_default(&space.id).await?;
```

Comment:
Maybe we could add the `set_X_as_default` call inside the `create_X` function, so we just use the `set_X_as_default` function when we want to explicitly change the default resource.

Reply:
This line is indeed redundant. When we create the first projet / space / identity etc... we set it as the default in `CliState::create_xxx`.